hbase-client/src/test/java/org/apache/hadoop/hbase/util/TestRoundRobinPoolMap.java

```diff
@@ -45,58 +52,104 @@ protected PoolType getPoolType() {
   }
 
   @Test
-  public void testSingleThreadedClient() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
-    String randomKey = String.valueOf(rand.nextInt());
-    String randomValue = String.valueOf(rand.nextInt());
-    // As long as the pool is not full, we'll get null back.
-    // This forces the user to create new values that can be used to populate
-    // the pool.
-    runThread(randomKey, randomValue, null);
-    assertEquals(1, poolMap.size(randomKey));
+  public void testGetOrCreate() throws IOException {
+    String key = "key";
+    String value = "value";
+    String result = poolMap.getOrCreate(key, () -> value);
+
+    assertEquals(value, result);
+    assertEquals(1, poolMap.values().size());
+  }
+
+  @Test
+  public void testMultipleKeys() throws IOException {
+    for (int i = 0; i < KEY_COUNT; i++) {
+      String key = Integer.toString(i);
+      String value = Integer.toString(2 * i);
+      String result = poolMap.getOrCreate(key, () -> value);
+
+      assertEquals(value, result);
+    }
+
+    assertEquals(KEY_COUNT, poolMap.values().size());
+  }
+
+  @Test
+  public void testMultipleValues() throws IOException {
+    String key = "key";
+
+    for (int i = 0; i < POOL_SIZE; i++) {
+      String value = Integer.toString(i);
+      String result = poolMap.getOrCreate(key, () -> value);
+
+      assertEquals(value, result);
+    }
+
+    assertEquals(POOL_SIZE, poolMap.values().size());
   }
 
   @Test
-  public void testMultiThreadedClients() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
+  public void testRoundRobin() throws IOException {
+    String key = "key";
+
     for (int i = 0; i < POOL_SIZE; i++) {
-      String randomKey = String.valueOf(rand.nextInt());
-      String randomValue = String.valueOf(rand.nextInt());
-      // As long as the pool is not full, we'll get null back
-      runThread(randomKey, randomValue, null);
-      // As long as we use distinct keys, each pool will have one value
-      assertEquals(1, poolMap.size(randomKey));
+      String value = Integer.toString(i);
+      poolMap.getOrCreate(key, () -> value);
     }
-    poolMap.clear();
-    String randomKey = String.valueOf(rand.nextInt());
-    for (int i = 0; i < POOL_SIZE - 1; i++) {
-      String randomValue = String.valueOf(rand.nextInt());
-      // As long as the pool is not full, we'll get null back
-      runThread(randomKey, randomValue, null);
-      // since we use the same key, the pool size should grow
-      assertEquals(i + 1, poolMap.size(randomKey));
+
+    assertEquals(POOL_SIZE, poolMap.values().size());
+
+    // pool is filled, get() should return elements round robin order
+    // starting from 1, because the first get was called by runThread()
```

Comment:
nit: this second line is wrong. the first value is now 0 and not 1, and runThread doesn't exist.